lib/graph/__init__.py

```diff
@@ -158,7 +158,7 @@ def append(self, node, edge=None):
         list.append(self, node)
 
     def remove(self, node):
-        if self._edges.has_key(node.id): del self._edges[node.id]
+        if node.id in self._edges: del self._edges[node.id]
         list.remove(self, node)
```

Comment:
Can all these deletion bits of code be fixed to use something atomic, like either -

```python
try:
    del self.edges[node.id]
except KeyError:
    pass
```

or 
`self.edges.pop(node.id, None)` ?

The whole of graph would probably improve readability with reformatting.